### PR TITLE
[Anomaly] CVS-98654 Add hazelnut dataset

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
# Description

- Adds hazelnut dataset as it is more realistic test

### CODE REVIEW

**Comment by ashwinvaidya17** [comment]
run ote-test

**Comment by ashwinvaidya17** [comment]
run ote-test

**Comment by ashwinvaidya17** [comment]
Here are the test results: http://validationreports.sclab.intel.com:8006/reports/build_number_report?test_session_build_number=pr-ote-test-829&environment=iotg-dev-workstation-08. Except for the openvino deploy tests, all other tests are passing. The deploy tests are failing on develop as well.

File: data/anomaly/Hazelnut_Toy/classification/test.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  "masks": {
+    "3": "mask/colour/06.png",
+    "4": "mask/colour/03.png",
+    "5": "mask/colour/07.png",
+    "6": "mask/colour/14.png",
+    "7": "mask/colour/01.png",
+    "8": "mask/colour/12.png",
+    "9": "mask/colour/11.png",
+    "10": "mask/colour/09.png",
+    "11": "mask/colour/08.png"
```
Comment (djdameln) [inline data/anomaly/Hazelnut_Toy/classification/test.json:39]: any reason why we only use colour defect type here, and not crack?

File: external/anomaly/adapters/anomalib/data/utils.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        y1 = comp_y / height
+        x2 = (comp_x + comp_w) / width
+        y2 = (comp_y + comp_h) / height
+
+        bboxes.append([x1, y1, x2, y2])
+
+    return bboxes
+
+
+def create_polygons_from_mask(mask_path: str) -> List[List[List[float]]]:
```
Comment (djdameln) [inline external/anomaly/adapters/anomalib/data/utils.py:47]: Return type is List[List[List[float]]] while the docstring says List[List[float]]

File: external/anomaly/adapters/anomalib/data/create_folder_json_annotations.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+import pandas as pd
+from adapters.anomalib.data.utils import (
+    create_bboxes_from_mask,
+    create_polygons_from_mask,
+    save_json_items,
+)
+from anomalib.data.folder import make_dataset
+
+
+def create_classification_json_items(pd_items: pd.DataFrame, data_root: str) -> Dict[str, Any]:
```
Comment (djdameln) [inline external/anomaly/adapters/anomalib/data/create_folder_json_annotations.py]: This code could be simplified. The contents of `create_classification_json_items`, `create_detection_json_items` and `create_segmentation_json_items` are largely the same, so I don't think we need separate functions for this.

Instead, we could use a single function where we also pass the task type as a parameter to determine if "masks" or "boxes" should be filled in the json.

**Comment by ashwinvaidya17** [comment]
I've addressed the comments

**Review by djdameln — APPROVED** [review]
Thanks!

**Comment by yunchu** [comment]
@ashwinvaidya17 Is it still valid PR?

**Comment by ashwinvaidya17** [comment]
@yunchu I wanted to address some performance issue by changing the dataset but it didn't really make a difference. Let's close this. I'll create a new one later on.

**Pull request closed without merging**